bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/dialogs/ProjectNaturesPage.java

```diff
@@ -300,7 +299,7 @@ protected void handle(InvocationTargetException e) {
 	 * Initializes a ProjectNaturesPage.
 	 */
 	private void initialize() {
-		project = (IProject) getElement().getAdapter(IResource.class);
+		project = getElement().getAdapter(IProject.class);
```

Comment:
Can you please try `Adapter.adapt(IProject.class)` or `Adatpter.adapt(IResource.class)` ?

Reply:
Sorry, but I don't understand why you want to adapt to IResource. This page is meant to adapt to IProject _only_. See https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/bundles/org.eclipse.ui.ide/plugin.xml#L858.
Or do you want to add `getAdapter(IResource.class).getProject()` instead, first adapting to IResource and then taking the containing project?

Reply:
Changed to `Adapter.getAdapter(IProject)`, since your concern was about `Adapter.get(...)`, as I assume from the similar comment on my other PR.

Reply:
I get it now. You care about the `Adapters` class, not the specific adapter. Fixed. Still have to add version bumping, missed that in the previous commit.